RuleDocumentation/UseProcessBlockForPipelineCommands.md

```diff
@@ -0,0 +1,62 @@
+# UseProcessBlockForPipelineCommands
+
+**Severity Level: Warning**
+
+## Description
+
+Functions that support pipeline input should always handle parameter input in a process block. Unexpected behavior can result if input is handled directly in the body of a function where parameters declare pipeline support.
+
+## Example
+
+### Wrong
+
+``` PowerShell
+Function Get-Number
+{
+	[CmdletBinding()]
+	Param(
+		[Parameter(ValueFromPipeline)]
+		[int]
+		$Number
+	)
+	
+	$Number
+}
+```
+
+#### Result
+
+```
+PS C:\> 1..5 | Broken
```

Comment:
I think you meant this?
```suggestion
PS C:\> 1..5 | Get-Number
```

Reply:
> The only last thing that came to my mind is that maybe `UseProcessBlockForPipelineCommand` might be a better name than `UseProcessBlockForPipelineCommands`, similar to the PowerShell design rule to use single nouns and user's will get a warning for a single command at a time and not commands. I do realize though that some of the other existing rules use a pluralized noun, therefore PSScriptAnalyzer rules already use a mix of styles where it wouldn't matter too much. I'leave that up to your judgement what you think/prefer, especially, since renaming the rule would also require a fair bit of work like renaming the files, etc.

That's fine - I'm always up for any improvements you can suggest. I'm making one more commit with your suggestions and the rule name change so hopefully this is all good now. Thanks again for your help with this, and sorry it took so long to get this up to snuff!

Reply:
Yep, sure did. Good catch! Resolved in 2df5e700a9ba117eed937ff6a715d5f177bd547a.